### ISSUE TITLE
Under panel 1.2.1, setting .visible to True for a widget that is initially created with visible=False doesn't toggle its visiblity

### ISSUE THREAD

**Issue by giftculture** [holoviz/panel#5354]
Thanks for contacting us! Please read and follow these instructions carefully, then delete this introductory text to keep your issue easy to read. Note that the issue tracker is NOT the place for usage questions and technical assistance; post those at [Discourse](https://discourse.holoviz.org) instead. Issues without the required information below may be closed immediately.

#### ALL software version info
Panel v1.2.1

#### Description of expected behavior and the observed behavior
 When creating a widget with visible=False, setting .visible to True doesn't reveal the widget.  The below code works under panel v 1.2.0 as expected - it reveals the spinner, sleeps for 5 seconds, and then hides the spinner.    Looking at the javascript console when the script is executed under panel v1.2.0, the css style ":host { display: none !important; }" is applied at the start, is removed when I click the submit button and .visible is set to true, and then added back once .visible is set to false

#### Complete, minimal, self-contained example code that reproduces the issue

```
import panel as pn
import logging
import bokeh
import time
logger = logging.getLogger('panel.tifviewer')
pn.extension(sizing_mode='stretch_width') #stretch_width


pn.__version__

load_and_display_spinner = pn.indicators.LoadingSpinner(value=False, height=100, width=100, visible=False)                      
submit = pn.widgets.Button(name='submit', button_type='primary')
main_ui = pn.Column(
                pn.Row(
                    pn.pane.Markdown(f'## my panel version is {pn.__version__}'),
                ),
                pn.Row(
                    submit
                ),
                pn.Row(
                    load_and_display_spinner
                ),
              
          ) 

tabs = pn.Tabs(
    ('app', main_ui),
).servable()
    

def load_display(spinner, x):
    print(f"spinner is {spinner}")
    print(f'switch is {x}')
    if (x == 'on'):
        spinner.value = True
        spinner.visible = True
    if (x == 'off'):
        spinner.value = False
        spinner.visible = False

def do_some_task(event):
    load_display(load_and_display_spinner, 'on')
    time.sleep(5)
    load_display(load_and_display_spinner, 'off')
                

submit.on_click(do_some_task)
```

**Comment by ahuang11** [comment]
```python
import panel as pn

pn.extension(sizing_mode="stretch_width")  # stretch_width

def toggle_visible(event):
    ls.visible = not ls.visible

ls = pn.widgets.LoadingSpinner(visible=False, value=True)
bt = pn.widgets.Button(name="Click me", button_type="primary")
bt.on_click(toggle_visible)

pn.Column(ls, bt).servable()
```

So if `ls.visible = True` initially, it works, but not if initialized with `ls.visible = False`

I think the reason is `visibility: hidden` doesn't get dropped
<img width="1305" alt="image" src="https://github.com/holoviz/panel/assets/15331990/9ca552b0-d0f4-4246-9c3a-530aa42c641a">

**Comment by emcd** [comment]
I see a fix for this was slated for the `v1.2.2` [milestone](https://github.com/holoviz/panel/milestone/57) and `v1.2.2` was [released yesterday](https://pypi.org/project/panel/1.2.2/). Is this going to be retargeted for `v1.2.3`? This is a fairly high visibility bug, if you'll pardon the pun.
